frontend/src/hocs/withMapboxToken/MapboxToken.ts

```diff
@@ -41,30 +45,26 @@ export class MapboxToken {
    * only be fetched once per session.)
    */
   public static async get(): Promise<string> {
-    if (!MapboxToken.token) {
-      if (SessionInfo.current.userMapboxToken !== "") {
-        MapboxToken.token = SessionInfo.current.userMapboxToken
-      } else {
-        const { commandLine } = SessionInfo.current
+    const { commandLine, userMapboxToken } = SessionInfo.current
 
+    if (
+      !MapboxToken.token ||
+      MapboxToken.commandLine !== commandLine.toLowerCase()
+    ) {
+      if (userMapboxToken !== "") {
+        MapboxToken.token = userMapboxToken
+      } else {
         if (
           this.isItRunningLocal() &&
           commandLine.toLowerCase() === "streamlit hello"
```

Comment:
Move this logic into `SessionInfo`. This component shouldn't know about command lines.

For example, just create a property `isHello` in `SessionInfo` like this:

```
class SessionInfo {
  get isHello(): boolean {
    return this.commandLine === "streamlit hello"
  }
}
```